pairinteraction/CMakeLists.txt

```diff
@@ -121,7 +121,7 @@ target_link_libraries(picomplex PUBLIC Eigen3::Eigen)
 # Find MKL
 set(MKL_ARCH "intel64" CACHE STRING "MKL architecture")
 set(MKL_LINK "dynamic" CACHE STRING "MKL linkage")
-set(MKL_THREADING "sequential" CACHE STRING "MKL threading backend")
+set(MKL_THREADING "intel_thread" CACHE STRING "MKL threading backend")
```

Comment:
In the long run, it would be cool if we could make the optimizations work with Windows as well. `gnu_thread` would not work there, I guess. Is there an easy way to enable threading cross-platform? If not, I am fine with `gnu_thread`.

Reply:
I just looked into `MKLConfig.cmake`: If no threading option is selected, it tries to find a suitable multithreading option by itself. Maybe, it is best to remove `set(MKL_THREADING "sequential" CACHE STRING "MKL threading backend")` and rely on this mechanism?

Reply:
Hmm, neither removing this line nor using `gnu_thread` worked with the CI.

Reply:
I found the problem, but I don't have a solution yet. In the build log you see
```
-- Found MKL: /opt/intel/oneapi/compiler/latest/linux/compiler/lib/intel64_lin/libiomp5.so  
```
but that file is a symbolic link and the dynamic linker seems to not like that.

For `gnu_thread` the problem is a bug (or rather inadequacy) of `MKLConfig.cmake`. It inspects `CMAKE_C_COMPILER_ID` if that is `GNU`, but ignores `CMAKE_CXX_COMPILER`. So the simplest solution for that would probably be
```cmake
set(CMAKE_C_COMPILER_ID "${CMAKE_CXX_COMPILER_ID}")
set(MKL_THREADING "gnu_thread" CACHE STRING "MKL threading backend")
```